google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleSupportProvider.java

```diff
@@ -165,6 +128,48 @@ public static void addSupport(@NotNull AppEngineFlexibleFacet facet,
     setupDeploymentRunConfiguration(facet.getModule());
   }
 
+  /**
+   * Generates an app.yaml configuration file in the src/main/appengine directory. If an app.yaml
+   * already exists it will not overwrite the file.
+   */
+  private static void generateAppYaml(@NotNull AppEngineFlexibleFacet facet,
+      @NotNull ModifiableRootModel rootModel) {
+    Project project = facet.getModule().getProject();
+    FileTemplate appYamlTemplate = FileTemplateManager.getInstance(project)
+        .getInternalTemplate(AppEngineTemplateGroupDescriptorFactory.APP_YAML_TEMPLATE);
+
+    try {
+      VirtualFile virtualFile = rootModel.getContentRoots()[0].findFileByRelativePath("src/main");
+
+      if (virtualFile != null) {
+        PsiDirectory directory = PsiManager.getInstance(project)
+            .findDirectory(virtualFile);
+
+        if (directory != null) {
+          PsiDirectory appEngineDirectory;
+          try {
+            directory.checkCreateSubdirectory(CONFIG_DIR_NAME);
```

Comment:
Do you really need to create the directory like this? Isn't it automatically created by createFromTemplate?